pkgs/top-level/linux-kernels.nix

```diff
@@ -182,6 +182,8 @@ in {
       kernelPatches = [
         kernelPatches.bridge_stp_helper
         kernelPatches.request_key_helper
+        kernelPatches.rust-1-72-fix
```

Comment:
nit: I'm a bit sad we need those patches even if we don't use Rust in the kernel but I guess doing conditionals here is worse in terms of rebuilds.

Reply:
To use the patches was the consensus in an earlier version of this PR. It was deemed less bad than having to package the matching rustc/bindgen for each kernel version. Given that they only patch Rust code, I would say it's ok.

The situation should be temporary and during 2024 all of the patching should go away.

Reply:
> The situation should be temporary and during 2024 all of the patching should go away.

What is the source on that?
I want this to be merged, and I still think patches are the best idea (I wrote many of the patches for nixos-apple-silicon).
But since the linux kernel has its own fork of the alloc crate vendored in and requires very fine grained control, it will always use unstable features (which will not be stabilized so soon, or ever).
Having unstable features means the code is not generally forward-compatible.
I am not aware of a plan to change this, so I would be going in with the expectation that upgrading rust in the future is always going to involve tinkering with the kernel a bit (but I'm willing to do this / have been doing this in the last months).

Reply:
We want to eventually support a minimum version, but when that will actually happen is unknown -- discussions with upstream Rust are ongoing. There are some details at https://rust-for-linux.com/rust-version-policy and https://rust-for-linux.com/unstable-features (we also have a section about `alloc` there).